### PR TITLE
[#137344025]supplier service editing: straighten out framework-determined behaviours for edit & remove views

### PR DESCRIPTION
**Pull request by risicle**
Story https://www.pivotaltracker.com/story/show/137344025

The first thing done here is to significantly refactor the tests around the edit & remove views.

The views are then fixed to work properly cross-framework. Frameworks without an `edit_service` content manifest will not allow editing **or** removal by a supplier. Not this latter change in behaviour - previously the removal of a e.g. dos service was not enforced by the actual remove view. The behaviour of the edit page for a dos supplier was... undefined (probably a 500). These are now 404s.

### CODE REVIEW

File: tests/app/main/test_services.py [hunk 1]
```diff
@@ -303,15 +330,48 @@ def test_should_view_disabled_service_with_removed_message(
...
+    def test_should_not_view_other_suppliers_services(
+            self, data_api_client, supplier_service_editing_fw_params
+    ):
+        framework_slug, framework_name, framework_editable_services = supplier_service_editing_fw_params
+        self.login()
+        self._setup_service(
+            data_api_client, framework_slug, framework_name, service_status='published', service_belongs_to_user=False)
 
+        res = self.client.get('/suppliers/services/123')
+        if not framework_editable_services:
```
Comment (TheDoubleK) [inline tests/app/main/test_services.py]: Is this check necessary as we're expecting a 404 either way?

File: tests/app/main/test_services.py [hunk 2]
```diff
@@ -303,15 +330,48 @@ def test_should_view_disabled_service_with_removed_message(
...
+        framework_slug, framework_name, framework_editable_services = supplier_service_editing_fw_params
         self.login()
-        self._get_service(data_api_client, fixture_framework_slug_and_name, service_status='published')
+        self._setup_service(data_api_client, framework_slug, framework_name, service_status='published')
 
+        # NOTE two http requests performed here
         res = self.client.post('/suppliers/services/123/remove', follow_redirects=True)
+        if not framework_editable_services:
+            assert_equal(res.status_code, 404)
+            assert_is(data_api_client.update_service_status.called, False)
```
Comment (TheDoubleK) [inline tests/app/main/test_services.py]: Aren't we killing nose methods rather than introducing more?

Comment (risicle) [inline tests/app/main/test_services.py]: The nose-removal patch sits on top of this one (*I think*) so the nose-removal happens after this - I guess I just wanted to keep it symmetrical with the rest of the code as shown?

File: tests/app/main/test_services.py (hunk 1 above)
Comment (risicle) [inline tests/app/main/test_services.py]: Ah yes indeed well spotted, I flipped a coin when I came to this one - did I want to add an unnecessary clause or make it less obvious that this test follows the same basic flow as the others. Scanning the code by eye would bring up a "hold on he's missed something here" moment. But looking at it now it does look kinda stupid.

Still, I don't want to have to do a chain-rebase... :hear_no_evil:

Comment (risicle) [inline tests/app/main/test_services.py]: Hold on I need to rebase anyway. Will fix.

Comment (risicle) [inline tests/app/main/test_services.py]: (I think I also had a bit of a :-\ should this really be a 403 instead? moment)

**Pull request merged**